R/meteo_imgw_daily.R

```diff
@@ -205,22 +236,56 @@ meteo_imgw_daily_bp = function(rank,
         temp = tempfile()
         temp2 = tempfile()
         test_url(addresses_to_download[j], temp)
-        unzip(zipfile = temp, exdir = temp2)
-        file1 = paste(temp2, dir(temp2), sep = "/")[1]
-        data1 = imgw_read(translit, file1)
-        colnames(data1) = meta[[1]]$parameters
-        # remove status
-        if (status == FALSE) {
-          data1[grep("^Status", colnames(data1))] = NULL
-        }
 
+                d = tryCatch(expr = unzip(zipfile = temp, exdir = temp2), 
+                     warning = function(w) {
+                       env$logs = c(env$logs, 
+                                    paste("Warning: ", w$message, " ",
+                                          addresses_to_download[j], sep = ""))
+                       # try to read it with archive package:
+                       data = archive_read(temp, file = paste0("o_d_", sprintf("%02d", j), "_", catalog, ".csv"), format = "zip")
```

Comment:
This and the next few lines are repeated a few times in the package. Maybe it would be worth to create an internal function for it?

Reply:
Yes, and no. There are slight differences in "o_d" or "k_d" or other weirdos used in IMGW convention. I would leave it as is for now with an option to improve it in next iterations.